Sources/Plasma/PubUtilLib/plNetClient/plNetClientMgr.cpp

```diff
@@ -403,50 +403,36 @@ void plNetClientMgr::IUnloadNPCs()
 //
 void plNetClientMgr::UpdateServerTimeOffset(plNetMessage* msg)
 {
-    if ((hsTimer::GetSysSeconds() - fLastTimeUpdate) > 5)
-    {
-        fLastTimeUpdate = hsTimer::GetSysSeconds();
-
-        const plUnifiedTime& msgSentUT = msg->GetTimeSent();
-        if (!msgSentUT.AtEpoch())
-        {
-            double diff = plUnifiedTime::GetTimeDifference(msgSentUT, plUnifiedTime::GetCurrent());
+    if (!msg->GetHasTimeSent())
+        return;
+    if (msg->GetTimeSent().AtEpoch())
+        return;
 
-            if (fServerTimeOffset == 0)
-            {
-                fServerTimeOffset = diff;
-            }
-            else
-            {
-                fServerTimeOffset = fServerTimeOffset + ((diff - fServerTimeOffset) / ++fTimeSamples);
-            }
+    double localTime = hsTimer::GetSeconds();
+    if (localTime - fLastLocalTime < 1.0)
+        return;
 
-            DebugMsg("Setting server time offset to {f}", fServerTimeOffset);
-        }
-    }
+    fLastServerTime = msg->GetTimeSent();
+    fLastLocalTime = localTime;
 }
 
-void plNetClientMgr::ResetServerTimeOffset(bool delayed)
+void plNetClientMgr::ResetServerTimeOffset()
 {
-    if (!delayed)
-        fServerTimeOffset = 0;
-    fTimeSamples = 0;
-    fLastTimeUpdate = 0;
+    fLastServerTime.ToEpoch();
+    fLastLocalTime = 0.0;
 }
 
 //
 // return the gameservers time
 //
 plUnifiedTime plNetClientMgr::GetServerTime() const 
-{ 
-    if ( fServerTimeOffset==0 )     // offline mode or before connecting/calibrating to a server
+{
+    if (fLastServerTime.AtEpoch()) {
+        WarningMsg("WARNING: Someone asked for the server time, but we don't know it yet!");
```

Comment:
`plLoggable::WarningMsg` already adds a `WRN: ` prefix

```suggestion
        WarningMsg("Someone asked for the server time, but we don't know it yet!");
```